### PR TITLE
[Experiment] Remove transient store

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

We can do without the transient store if we are ok letting authenticators authors modify state on the ante.

I think that's ok as long as it *only* happens in track() which is run after authentication happens. This expectation needs to be explicitly documented for cw-authenticator authors

(drive-by, remove IsReady)

## Testing and Verifying
All tests should pass, only one test needs to be updated(the one that explicitly tests this

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/authenticator/ante/ante.go [hunk 1]
```diff
@@ -101,11 +101,10 @@ func (ad AuthenticatorDecorator) AnteHandle(
 		}
 	}()
 
-	// This is a transient context stored globally throughout the execution of the tx
-	// Any changes will to authenticator storage will be written to the store at the end of the tx
-	cacheCtx := ad.authenticatorKeeper.TransientStore.ResetTransientContext(ctx)
+	cacheCtx, writeCache := ctx.CacheContext()
+
 	// Ensure that no usedAuthenticators are stored in the transient store
```
Comment (iboss-ptk) [inline x/authenticator/ante/ante.go:106]: The comment still mention transient store

**Review by iboss-ptk — CHANGES_REQUESTED** [review]
The rest LGTM, we also want to [remove `wasmStorekey` here](https://github.com/osmosis-labs/osmosis/blob/6f79cd4aa55c5def878da4fd61d0ffdb135fa65b/x/authenticator/authenticator/cosmwasm.go#L34) as well since we no longer needs to sync cosmwasm state

**Review by PaddyMc — APPROVED** [review]
:mage_man:

File: x/authenticator/ante/ante.go (hunk 1 above)
Comment (nicolaslara) [reply x/authenticator/ante/ante.go:106]: The comment was removed.

**Pull request merged**